src/main/groovy/de/itemis/mps/gradle/CreateDmg.groovy

```diff
@@ -59,11 +73,19 @@ class CreateDmg extends DefaultTask {
                             'Mac/Finder/DSStore/BuddyAllocator.pm', 'Mac/Finder/DSStore.pm']
         File scriptsDir = File.createTempDir()
         File dmgDir = File.createTempDir()
+        def getSigningInfo = [signKeyChainPassword, signKeyChain, signIdentity]
         try {
             BundledScripts.extractScriptsToDir(scriptsDir, scripts)
             project.exec {
                 executable new File(scriptsDir, 'mpssign.sh')
-                args rcpArtifact, dmgDir, jdk
+
+                if(getSigningInfo.every {el -> el != null}) {
+                    args '-r', rcpArtifact, '-o', dmgDir, '-j', jdk, '-p', signKeyChainPassword, '-k', signKeyChain, '-i', signIdentity
+                }else if (getSigningInfo.every {el -> el == null}){
+                    args '-r', rcpArtifact, '-o', dmgDir, '-j', jdk
+                }else{
+                    throw new IllegalArgumentException("Not all signing paramters set:  signKeyChainPassword: ${getSigningInfo[0]}, signKeyChain: ${getSigningInfo[1]}, signIdentity: ${getSigningInfo[2]} ")
```

Comment:
We said we don't want to display the password to the user, so I wouldn't include it in the exception message either.